src/components/Profile/SettingsContainer/index.tsx

```diff
@@ -0,0 +1,147 @@
+import { View, Text } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { useNavigation } from '@react-navigation/native';
+import styles from './styles';
+import { AppContext } from '../../../contexts/AppContext';
+import { UserContext } from '../../../contexts/UserContext';
+
+import SettingsBox from '../SettingsBox';
+
+const SettingsContainer = () => {
+  const navigation = useNavigation();
+  const { setAppState } = useContext(AppContext);
+  const { userState, setUserState } = useContext(UserContext);
+
+  // workaround to parse data that may be undefined - to change?
+  const homes = userState.homes.items;
+  const homesString = JSON.stringify(homes);
+  const homesJSON = JSON.parse(homesString);
+  const { home } = homesJSON[0];
+  const curHomeName = homesJSON[0].home.addressLine1.split(' ')[1];
+
+  const [homeData] = useState(home);
+  const [name, setName] = useState(String(userState.displayName));
+  const [email, setEmail] = useState(String(userState.username));
+  const [password, setPassword] = useState('');
+  const [address, setAddress] = useState(
+    String(homeData.addressLine1) === 'null'
+      ? ''
+      : String(homeData.addressLine1),
+  );
+  const [address2, setAddress2] = useState(
+    String(homeData.addressLine2) === 'null'
+      ? ''
+      : String(homeData.addressLine2),
+  );
+  const [homeName, setHomeName] = useState(curHomeName);
+
+  const signOut = async () => {
+    try {
+      await Auth.signOut();
+      setUserState({
+        _deleted: null,
+        _lastChangedAt: 0,
+        _version: 1,
+        createdAt: '',
+        displayName: '',
+        homes: {
+          items: [],
+          nextToken: null,
+          startedAt: null,
+        },
+        id: '',
+        owner: '',
+        updatedAt: '',
+        username: '',
+      });
+      setAppState('Auth');
+    } catch (error) {
+      console.error('Error signing out: ', error);
+    }
+  };
+
+  const onboardingPage = () => {
+    navigation.navigate('Onboarding');
+  };
+
+  return (
+    <View>
+      {/* account section */}
+      <View style={styles.account}>
+        <Text style={styles.title}>ACCOUNT</Text>
+        {/* table */}
+        <SettingsBox
+          type="Input"
+          icon="person-circle-outline"
+          place="Name"
+          valueInput={name}
+          setInput={setName}
+        />
+        <SettingsBox
+          type="Input"
+          icon="mail-outline"
+          place="Email"
+          valueInput={email}
+          setInput={setEmail}
+        />
+        <SettingsBox
+          type="Input"
+          icon="lock-closed-outline"
+          place="Password"
+          valueInput={password}
+          setInput={setPassword}
+          password
+        />
+        <SettingsBox
+          type="Input"
+          icon="home-outline"
+          place="Current Address"
+          valueInput={address}
+          setInput={setAddress}
```

Comment:
In these text boxes are where you'd want to execute the mutation then context updates.